packages/blockly/src/components/custom-block-form-config.ts

```diff
@@ -19,19 +19,21 @@ export interface IBlockTypeConfig {
   optionTerm?: string;
   optionValuePlaceholder?: string;
   placeholder?: string;
+  typeCanHaveChildren?: boolean;
```

Comment:
I think I understand the need for this -- the existing `canHaveChildren` only indicates whether the option to nest child blocks has been activated, while `typeCanHaveChildren` indicates whether the type of block supports nesting child blocks. Is that right?

If so, I wonder if we should adjust the names to help prevent confusion around that? Maybe something like `childrenEnabled` and `supportsChildren`? Or `nestingEnabled` and `supportsNesting`?